Nightly build failures with gcc/9.2.0 with c++20 in TestAtomicOperations_shared.hpp: error: implicit capture of ‘this’ via ‘[=]’ is deprecated in C++20

gcc/9.2.0 builds with c++20 began failing to compile with recent changes with error message:
```
22:26:51 cd /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/TestAll_2021-12-14_22.25.02/gcc/9.2.0/Serial-release/core/unit_test && /projects/sems/install/rhel7-x86_64/sems/compiler/gcc/9.2.0/base/bin/g++  -DGTEST_HAS_PTHREAD=0 -DGTEST_HAS_TR1_TUPLE=0 -DKOKKOS_DEPENDENCE -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/tpls/gtest -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/TestAll_2021-12-14_22.25.02/gcc/9.2.0/Serial-release/core/unit_test -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/core/unit_test -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/core/unit_test/category_files -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/TestAll_2021-12-14_22.25.02/gcc/9.2.0/Serial-release -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/TestAll_2021-12-14_22.25.02/gcc/9.2.0/Serial-release/core/src -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/core/src -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/TestAll_2021-12-14_22.25.02/gcc/9.2.0/Serial-release/containers/src -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/containers/src -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/TestAll_2021-12-14_22.25.02/gcc/9.2.0/Serial-release/algorithms/src -I/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/algorithms/src  -O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized -O3 -DNDEBUG   -mavx -std=c++2a -o CMakeFiles/KokkosCore_UnitTest_Default.dir/UnitTestMainInit.cpp.o -c /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/core/unit_test/UnitTestMainInit.cpp
22:26:51 In file included from /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/TestAll_2021-12-14_22.25.02/gcc/9.2.0/Serial-release/core/unit_test/serial/TestSerial_AtomicOperations_shared.cpp:2:
22:26:51 /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/core/unit_test/TestAtomicOperations_shared.hpp: In lambda function:
22:26:51 /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/core/unit_test/TestAtomicOperations_shared.hpp:66:40: error: implicit capture of ‘this’ via ‘[=]’ is deprecated in C++20 [-Werror=deprecated]
22:26:51    66 |     Kokkos::single(Kokkos::PerTeam(t), [=]() { m_view() = *x; });
22:26:51       |                                        ^
22:26:51 /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_gcc920_openmp_serial_cpp20/kokkos/core/unit_test/TestAtomicOperations_shared.hpp:66:40: note: add explicit ‘this’ or ‘*this’ capture
```

**Reproducer** (serial backend):
```
module load sems-env sems-cmake/3.17.1 sems-gcc/9.2.0

$KOKKOS_PATH/generate_makefile.bash --with-devices=Serial --arch=SNB --compiler=g++ --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized " --cxxstandard="20" --ldflags=""  --kokkos-path=$KOKKOS_PATH --with-options=disable_deprecated_code
```

--- a/core/unit_test/TestAtomicOperations_shared.hpp
+++ b/core/unit_test/TestAtomicOperations_shared.hpp
@@ -59,11 +59,11 @@ struct TestSharedAtomicsFunctor {
   KOKKOS_INLINE_FUNCTION void operator()(
       const typename Kokkos::TeamPolicy<ExecutionSpace>::member_type t) const {
     int* x = (int*)t.team_shmem().get_shmem(sizeof(int));
-    Kokkos::single(Kokkos::PerTeam(t), [=]() { *x = 0; });
+    Kokkos::single(Kokkos::PerTeam(t), [&]() { *x = 0; });
     t.team_barrier();
     Kokkos::atomic_add(x, 1);
     t.team_barrier();
-    Kokkos::single(Kokkos::PerTeam(t), [=]() { m_view() = *x; });
+    Kokkos::single(Kokkos::PerTeam(t), [&]() { m_view() = *x; });
   }
 };
 

